packages/block-library/src/navigation-link/edit.js

```diff
@@ -129,6 +131,43 @@ function NavigationLinkEdit( {
 		selection.addRange( range );
 	}
 
+	function handleCreateEntity( type, entityTitle ) {
+		return saveEntityRecord( 'postType', type, {
+			title: entityTitle,
+			status: 'publish',
+		} ).then( ( entity ) => {
+			// `entity` may not reject the Promise
+			// but may still be invalid. Here we
+			// tests for unexpected values and throw accordingly.
+			if ( null === entity || undefined === entity ) {
+				throw new TypeError(
+					'API response returned invalid entity.',
+					entity
+				);
+			}
+
+			const requiredEntityProps = [ 'id', 'title', 'link' ];
+
+			const entityMissingProperty = requiredEntityProps.find(
+				( entityProp ) => ! entity.hasOwnProperty( entityProp )
+			);
+
+			if ( Boolean( entityMissingProperty ) ) {
```

Comment:
The `Boolean` cast isn't really necessary, since `if` would evaluate the value as truthy anyways.

```suggestion
			if ( entityMissingProperty ) {
```

Reply:
Yep. I understand the coercion rules that apply here. Indeed I've been making something of a study of this recently.

I felt that as `.find` could return `undefined` or the "found" value it would be clearer to explicitly cast to a `Boolean`. I guess this doesn't really conform to the style guide, however.